imports/importers.py

```diff
@@ -48,49 +67,116 @@ class ShapefileImporter:
     new ones.
     """
 
+    error_messages = {
+        "fields_not_allowed": _(
+            "Fields not allowed: %(fields)s. Allowed shapefile fields are: %(allowed_fields)s."
+        ),
+        "feature_classes_not_exist": _(
+            "Feature classes do not exist: %(feature_classes)s."
+        ),
+    }
+
     @classmethod
     def import_features(cls, zipped_shapefiles):
-        shp_reader = cls._get_shp_reader(zipped_shapefiles)
-        fields = [
-            SHAPEFILE_FIELD_MAPPING[shape_field[0]]
-            for shape_field in shp_reader.fields[1:]
-        ]
-        for shape_record in shp_reader.iterShapeRecords():
-            cls._import_feature(fields, shape_record)
-        return shp_reader.numRecords
+        shapefile_importer = cls()
+        shapefiles = shapefile_importer._extract_shapefiles(zipped_shapefiles)
+        data_source = DataSource(shapefiles)
+        layer = data_source[0]  # shapefiles always have one layer
+        shapefile_importer._validate_layer(layer)
+        return shapefile_importer._import_layer(layer)
 
-    @classmethod
-    def _get_shp_reader(cls, zipped_shapefiles):
-        with zipfile.ZipFile(zipped_shapefiles) as zfile:
-            dbf, shp, shx = sorted(
-                [
-                    name
-                    for name in zfile.namelist()
-                    if name.endswith(REQUIRED_SHAPEFILE_EXTENSIONS)
-                ]
+    def _extract_shapefiles(self, zipped_shapefiles):
```

Comment:
This is called "extract shapefile**s**" and while that is true that this _would_ extract multiple files if such existed, it still handled only one such file and also just return a single file path.
Minor nitpick, however. 

Overall, isn't .shp a shapefile, not plural? Just checked Wikipedia and it seems to call it that at least. But again, nitpicking here really.

Reply:
This is an oversight, I think we should name it in the same way as the class and use singular form here.

And just some clarification, the term `shapefile` can be confusing as it cab refer to the collection of files that together forms a single shapefile. Although we only pass `.shp` file path to the DataSource (and then to GDAL API inside DataSource), all files are actually used by GDAL when parsing the shapefile. GDAL will try to find other files based on the `.shp` file path.